### ISSUE TITLE
[Sanity] Create a schema plan

### ISSUE THREAD

**Issue by dvd-lara** [UPE-FIU/UPE-Site#5]
### Description
Please research Sanity and how their schema system works. A good place to get started is [here](https://www.sanity.io/docs/getting-started). For this issue, please collaborate with the other back-end members of the team to devise a plan for the schema. 

### Some objects the website will need (feel free to add more and discuss which are necessary)
- Images (containing a title field, an image field, and an alt text field)
- Events (containing a title field, a program field (the program that is hosting the event), an image field (for flyers))
- Pages (this might be the trickiest since page needs will vary. refer to the mockup for what each type of page will need)

**Comment by byfnoel** [comment]
Would **participants** be needed as an object? This would hold participand_Id, teams, and programs.

**Comment by byfnoel** [comment]
Thinking of a **participants** table

Would it be necessary?

**Comment by dvd-lara** [comment]
> 
> 
> Thinking of a **participants** table
> 
> Would it be necessary?

I don't believe it would be necessary at this moment. The schema plan should be for the minimum viable version of the website which simply displays information about the club, the programs and their events

**Comment by byfnoel** [comment]
Understood!

**Comment by jhern603** [comment]
Here is an image with two possible schema designs.
![image](https://user-images.githubusercontent.com/43681757/132605593-10da180b-1b27-4a87-8e98-34048e872c8e.png)

Ideally, we want to minimize the number of queries we make on the front-end so that the page has faster performance. One thing we will have to look into is lazy loading. I personally don't have much experience lazy loading, so this will be a learning experience for all of us. 

What we want happening in the back-end is to be relating the different fields of the tables together, so that the front-end can achieve a minimum number of queries.

An example of this:

If you look at the schema in the bottom right of the image, the **Program** table may pull the title and the flyer for events from the **Event** table.

Ideally, this would make the query from the front end look like this:

(Query everything from the Event table)
*[_type == 'Event']
(Only display the Title, and the Flyer for each event)
*[_type == 'Event']{Title, Flyer}

**SQL Equivalent**: _SELECT_ Title, Flyer _FROM_  _schema _WHERE_ type = 'Event'

We will have to make an experimental schema so we can see if cross-referencing is really this easy, though.

Once we agree on a schema design, and verify that cross-referencing is viable, I will design an ERD so we have a clear graphic on what the schema will look like.

**Comment by ZiadMalik1** [comment]
Based off my Understanding of Schemas, I feel that the more Compact and efficient they are, the more they serve their purpose of keeping organization. **The Bottom Right Schema seems to be the one that in my opinion, is perfect for the job.** Using SK and PK, we can create Queries that pull off of one another in different tables.
